### ISSUE TITLE
Large flashing coronas appear when looking at shiny things e.g. mirrors

### ISSUE THREAD

**Issue by iamgreaser** [Neos-Metaverse/NeosPublic#2620]
Context: I'm on Linux and running this in desktop mode.

## Describe the bug
Coronas come in two different flavours: Really really large and bright, or nonexistent. The exact threshold for this is not well-determined, as you can be looking at an angle and get a corona, then the slightest nudge in any direction makes it disappear.

... I'm not sure if corona is an entirely accurate term but it seems to be the graphics term for putting a texture on the screen to make something look shiny (not to be confused with a reflection map) - this seems to be the closest definition that I can find on Wikipedia: [Corona \(optical phenomenon\)](https://en.wikipedia.org/wiki/Corona_\(optical_phenomenon\))

## Screenshots / Video
![image](https://user-images.githubusercontent.com/138113/125181972-1611ca80-e25e-11eb-98e9-678a13d8814d.png)

## To Reproduce
This generally happens any time you try to use a mirror. It can also happen when editing LogiX stuff but I'm not sure how to make it happen consistently.

It's something I first noticed on a mirror in the first suggested avatar world from the tutorial, and it tends to happen on all mirrors I use in practice outside of a basic home world.

Creating a mirror and then putting a point light source in front of it is enough to reproduce the bug.

## Expected behavior
If something is so shiny that it makes a spot this bright (see screenshot), I would expect to move my mouse around a bit and that spot would get smaller instead of flickering as I move.

## Bug information (please complete the following information):
 - How often does it happen: Always
 - Does the bug persist after restarting Neos? Yes
 - Neos Version: Beta 2021.7.9.1297
 - Neos Platform: Linux
 - Regression: No

## Additional context
I've tested this on both my AMD Radeon RX 570 and my integrated Intel HD 530. Both have the same problem.

I attempted to test the Windows version using Proton 5.0 for comparison (later versions have issues w/ audio simply not working here) but it crashed in the loading screen, unfortunately.

## Reporters:
Just myself (GreaseMonkey).

**Comment by iamgreaser** [comment]
Something to note before I start, #2072 is a concrete case of this.

It turns out that on Linux, the graphics drivers let you dump shaders and put replacement shaders in: https://docs.mesa3d.org/shading.html

There are 3 shaders which refer to a `sampler2D` of `_BloomTex`.

If we replace all `vec4(0.0)`, then the coronas disappear. Of course, all bloom disappears and stuff looks quite nice, which is making it hard for me to want to narrow down the issue.

But let's narrow it down anyway. If we only disable the samplings of the bloom texture in one shader and leave the rest alone, the coronas stop being an issue... but also there's still no bloom.

The 3 shaders I'm looking at, on my machine, have the following line counts (this may vary on your machine but they'll probably keep their relative sizes):

- 81 lines: Samples `_BloomTex` once.
- 129 lines: Samples `_BloomTex` 9 times. **This is the shader where the coronabugs are showing up.** But it also covers bloom in general.
- 167 lines: Samples `_BloomTex` 9 times. I suspect this is a variant of the 129-line shader, but I'm not sure when this shader actually gets used.

So basically, the bug pertains to how bloom is implemented, and its effects show up in at least one shader or shader variant.

Bloom in general works fine. But certain cases of it just... break horribly, and then you end up with coronabugs.

I will stress two things:

- The shader I have located is almost certainly not the cause of the bug.
- But disabling the bloom in the shader does make Neos look nicer.

I'll look  further into this, but I will classify having bloom completely disabled as a fix for this bug.

**Comment by ghost** [comment]
I don't really have the issue on mirrors, but I do have the issue.  (If I have the issue and there is a mirror then it will probably be in the mirror though.)

**Comment by Psychpsyo** [comment]
Since no one has confirmed this here yet:
This does not happen under Windows on my machine but does happen under Linux. (Radeon RX590)

Whether or not it happens in mirrors seems to be inconsistent on my end and I haven't tried narrowing that down yet.
